FireRoad/CourseSearchEngine.swift

```diff
@@ -372,7 +406,18 @@ class CourseSearchEngine: NSObject {
                 } else {
                     relevance *= log(Float(max(2, course.enrollmentNumber)))
                 }
-                newResults[course] = relevance
+                let pattern = try! NSRegularExpression(pattern: "(\\d+|\\S+\\.)(\\.?\\S*)", options: .caseInsensitive)
+                if pattern.numberOfMatches(in: searchTerm, options: [], range: NSRange(location: 0, length: searchTerm.count)) > 0 {
```

Comment:
You're probably still working on this, but maybe we want to check if the entire search term matches instead of just a part? We could do that by adding `^` to the beginning and `$` to the end of the regex pattern (matches the beginning and end of the string, respectively). Also, per our discussion last week, we probably want to do a prefix search if the search term contains a `.` character, and otherwise a simple contains search.